runtime/command-metrics/src/main/java/io/aklivity/zilla/runtime/command/metrics/internal/airline/ZillaMetricsCommand.java

```diff
@@ -128,67 +78,68 @@ private Path engineDirectory()
-    {
-        return path.getNameCount() - metricsPath.getNameCount() == 1 &&
-                GAUGES_PATTERN.matcher(path.getName(path.getNameCount() - 1).toString()).matches() &&
-                Files.isRegularFile(path);
-    }
-
-    private boolean isHistogramsFile(
-        Path path)
-    {
-        return path.getNameCount() - metricsPath.getNameCount() == 1 &&
-                HISTOGRAMS_PATTERN.matcher(path.getName(path.getNameCount() - 1).toString()).matches() &&
-                Files.isRegularFile(path);
+        do
+        {
+            try (Engine engine = Engine.builder()
+                .config(config)
+                .errorHandler(onError)
+                .readonly()
+                .build())
+            {
+                engine.start();
+                MetricsReader metrics = new MetricsReader(engine, engine::supplyLocalName);
+                LongPredicate filterPredicate = filterPredicate(namespace, binding, engine::supplyLabelId);
+                List<MetricRecord> records = filter(metrics.records(), filterPredicate);
+                MetricsPrinter printer = new MetricsPrinter(records);
+                printer.print(System.out);
+            }
+            catch (Throwable ex)
+            {
+                System.out.println("error");
+                rethrowUnchecked(ex);
+            }
+            sleep(interval);
+        } while (interval != 0);
     }
 
-    private CountersLayout newCountersLayout(
-        Path path)
+    private LongPredicate filterPredicate(
+        String namespace,
+        String binding,
+        Function<String, Integer> lookupLabelId)
     {
-        return new CountersLayout.Builder().path(path).mode(READ_ONLY).build();
+        int namespaceId = namespace != null ? Math.max(lookupLabelId.apply(namespace), 0) : 0;
+        int bindingId = binding != null ? Math.max(lookupLabelId.apply(binding), 0) : 0;
+        long namespacedId = (long) namespaceId << Integer.SIZE | (long) bindingId << 0;
+        long mask =
+            (namespace != null ? 0xffff_ffff_0000_0000L : 0x0000_0000_0000_0000L) |
+                (binding != null ? 0x0000_0000_ffff_ffffL : 0x0000_0000_0000_0000L);
+        LongPredicate filter = id -> (id & mask) == namespacedId;
+        return filter;
     }
 
-    private GaugesLayout newGaugesLayout(
-        Path path)
+    private List<MetricRecord> filter(
+        List<MetricRecord> records,
+        LongPredicate filterPredicate)
     {
-        return new GaugesLayout.Builder().path(path).mode(READ_ONLY).build();
+        return records.stream()
+            .filter(r -> filterPredicate.test(r.bindingId()))
+            .collect(Collectors.toList());
     }
 
-    private HistogramsLayout newHistogramsLayout(
-        Path path)
+    private void sleep(
+        long interval)
     {
-        return new HistogramsLayout.Builder().path(path).mode(READ_ONLY).build();
+        try
+        {
+            Thread.sleep(interval * 1000L);
```

Comment:
Lets precompute interval to the value we need (millis) instead of recalculating it each time.

Reply:
done